config/v1/tests/ingresses.config.openshift.io/SetEIPForNLBIngressController.yaml

```diff
@@ -0,0 +1,224 @@
+tests:
+  onCreate:
+    - name: Should not allow to set NLB parameters when LBType is Classic.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: Classic
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-12345
+                  - eipalloc-12346
+      expectedError: "Network load balancer parameters are allowed only when load balancer type is NLB."
+    - name: Should allow to set NLB parameters when LBType is NLB.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: NLB
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-1234567891011abcd
+                  - eipalloc-1234695789abcdefa
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+                type: AWS
+                aws:
+                  type: NLB
+                  networkLoadBalancer:
+                    eipAllocations:
+                    - eipalloc-1234567891011abcd
+                    - eipalloc-1234695789abcdefa
+    - name: Should not be able to create default ingresscontroller if the value of eipAllocations is not unique.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: NLB
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-12345
+                  - eipalloc-12345
+      expectedError: "eipAllocations: Invalid value: \"array\": eipAllocations cannot contain duplicates"
+    - name: Should not be able to create default ingresscontroller if the value of eipAllocations is more than 26 characters.
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Ingress
+        metadata:
+          name: cluster
+        spec:
+          loadBalancer:
+            platform:
+              type: AWS
+              aws:
+                type: NLB
+                networkLoadBalancer:
+                  eipAllocations:
+                  - eipalloc-123450000000000000000000000000000
+      expectedError: "[spec.loadBalancer.platform.aws.networkLoadBalancer.eipAllocations[0]: Too long: may not be longer than 26"
+    - name: Should not be able to create default ingresscontroller if the value of eipAllocations is less than 26 characters.
```

Comment:
Same here and below—looks like copypasta?  The tests in this file are all for the cluster config object, so they should refer to that.